lte/gateway/c/oai/tasks/grpc_service/HaServiceImpl.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <string>
+
+#include "HaServiceImpl.h"
+#include "lte/protos/ha_service.pb.h"
+extern "C" {
+#include "log.h"
+#include "intertask_interface.h"
+#include "intertask_interface_types.h"
+#include "itti_types.h"
+}
+extern task_zmq_ctx_t grpc_service_task_zmq_ctx;
+
+namespace magma {
+using namespace lte;
+
+HaServiceImpl::HaServiceImpl() {}
+/*
+ * StartAgwOffload is called by North Bound to release UEs
+ * based on eNB identification or UE identification.
+ */
+grpc::Status HaServiceImpl::StartAgwOffload(
+    grpc::ServerContext* context, const StartAgwOffloadRequest* request,
+    StartAgwOffloadResponse* response) {
+  OAILOG_INFO(LOG_UTIL, "Received StartAgwOffloadRequest GRPC request\n");
+  send_agw_offload_req(request);
```

Comment:
Will there be other calls to `send_agw_offload_req` from other functions? If not, we can just include the logic here. All StartAgwOffload is doing is call `send_agw_offload_req`.

Reply:
indeed!